lint/src/test/java/permissions/dispatcher/PermissionDispatcherDetectorTest.java

```diff
@@ -0,0 +1,84 @@
+package permissions.dispatcher;
+
+import com.android.tools.lint.checks.infrastructure.LintDetectorTest;
+import com.android.tools.lint.client.api.LintClient;
+import com.android.tools.lint.detector.api.Detector;
+import com.android.tools.lint.detector.api.Issue;
+import com.android.tools.lint.detector.api.Project;
+
+import org.junit.Test;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class PermissionDispatcherDetectorTest extends LintDetectorTest {
+
+    private Set<Issue> mEnabled = new HashSet<>();
+
+    @Override
+    protected Detector getDetector() {
+        return new PermissionDispatcherDetector();
+    }
+
+    @Override
+    protected List<Issue> getIssues() {
+        return Collections.singletonList(PermissionDispatcherDetector.ISSUE);
+    }
+
+    @Override
+    protected TestConfiguration getConfiguration(LintClient client, Project project) {
+        return new TestConfiguration(client, project, null) {
+            @Override
+            public boolean isEnabled(Issue issue) {
+                return super.isEnabled(issue) && mEnabled.contains(issue);
+            }
+        };
+    }
+
+    @Override
+    protected boolean allowCompilationErrors() {
+        return true;
+    }
+
+    @Override
+    protected InputStream getTestResource(String relativePath, boolean expectExists) {
+        String path = "lint/src/test/java/permissions/dispatcher/data/".replace("/", File.separator) + relativePath;
```

Comment:
On a side note, I'm running into the same issues again, even though the project claims to have fixed the issues with `getResourceAsStream()`.